pkgs/by-name/wi/wifi-qr/package.nix

```
@@ -0,0 +1,81 @@
+{ lib
+, fetchFromGitHub
+, installShellFiles
+, makeWrapper
+, gnome
+, networkmanager
+, qrencode
+, stdenvNoCC
+, xdg-utils
+, zbar
+}:
+stdenvNoCC.mkDerivation {
+  pname = "wifi-qr";
+  version = "unstable-2023-09-30";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromGitHub {
+    owner = "kokoye2007";
+    repo = "wifi-qr";
+    rev = "821892001f735dc250a549ea36329cdc767db9c9";
+    hash = "sha256-kv0qjO+wn4t//NmKkHB+tZB4eRNm+WRUa5rij+7Syuk=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ];
+
+  dontBuild = true;
+
+  dontConfigure = true;
+
+  postPatch = ''
+    substituteInPlace src/wifi-qr.desktop \
+      --replace "Exec=sh -c 'wifi-qr -g'" "Exec=$out/bin/wifi-qr -g" \
+      --replace "Exec=sh -c 'wifi-qr -q'" "Exec=$out/bin/wifi-qr -q" \
+      --replace "Exec=sh -c 'wifi-qr -p'" "Exec=$out/bin/wifi-qr -p" \
+      --replace "Exec=sh -c 'wifi-qr -c'" "Exec=$out/bin/wifi-qr -c" \
+      --replace "Icon=wifi-qr.svg" "Icon=wifi-qr"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 src/wifi-qr $out/bin/wifi-qr
+
+    install -Dm644 src/wifi-qr.desktop $out/share/applications/wifi-qr.desktop
+    install -Dm644 src/wifi-qr.svg $out/share/icons/hicolor/scalable/apps/wifi-qr.svg
+
+    installManPage src/wifi-qr.1
+
+    runHook postInstall
+  '';
+
+  wrapperPath = lib.makeBinPath [
+    gnome.zenity
+    networkmanager
+    qrencode
+    xdg-utils
+    zbar
+  ];
+
+  fixupPhase = ''
+    runHook preFixup
+
+    patchShebangs $out/bin/wifi-qr
+    wrapProgram $out/bin/wifi-qr --suffix PATH : "$wrapperPath"
+
+    runHook postFixup
+  '';
```

Consider using `patsh` instead.

```suggestion
  postInstall = ''
    patsh -f $out/bin/wifi-qr -s ${builtins.storeDir}
  '';
```

The inputs must be added to `buildInputs`

---

I used `fixupPhase` because that seems more appropriate IMO.

I tried adding `bash` to `buildInputs` but it looks like `patsh` doesn't fix the shebang, so `patchShebangs` stays.